src/lib/util/request.ts

```diff
@@ -0,0 +1,55 @@
+import type { IncomingMessage } from "http";
+import { parseLanguageTag } from "@sozialhelden/ietf-language-tags";
+import type { ILanguageSubtag } from "@sozialhelden/ietf-language-tags";
+import { type ParsedQuery, parseUrl } from "query-string";
+import { parseAcceptLanguageString } from "~/lib/i18n/parseAcceptLanguageString";
+
+export function getRequestUrl(request?: IncomingMessage): string {
+  return request?.url ?? window.location.href;
+}
+
+export function getRequestUserAgentString(request?: IncomingMessage): string {
+  return request?.headers["user-agent"] ?? navigator.userAgent;
+}
+
+export function getRequestHostname(request?: IncomingMessage): string {
+  const withPort = request?.headers.host ?? window.location.hostname;
+  return withPort.split(":")[0];
+}
+
+export function getRequestQuery(
+  request?: IncomingMessage,
+): ParsedQuery<string> {
+  return parseUrl(getRequestUrl(request)).query;
+}
+
+export function getRequestLanguageTags(request?: IncomingMessage): string[] {
+  if (request) {
+    return parseAcceptLanguageString(
+      request.headers?.["accept-language"] ?? "",
```

Comment:
While most browsers send an `Accept-Language` header, our liveness probe does not, which likely is the reason why https://refactor-app-context.wheelmap.tech isn't reachable.

I guess this fails if there is no `accept-language` header. I'd suggest to fall back to 'English (US)' like this (or using the central constant):

```suggestion
      request.headers?.["accept-language"] ?? "en-US",
```

…or to simply return `'en-US'` in the function if the header is missing.